File: anmol JS/webcam.js
var state = 0;

//set up client to talk and listen to IoE registration server
function setup() {
	
	IoEClient.setup({
		type: "Webcam",
		states: [{
			name: "On",
			type: "bool",
			controllable: true
		},
		{
			name: "Image",
			type: "image"
		}]
	});
	
	IoEClient.onInputReceive = function(input) {
		processData(input, true);
	};
	
	attachInterrupt(0, function() {
		processData(customRead(0), false);
	});
	
	state = restoreProperty("state", 0);
	sendReport();
}

function restoreProperty(propertyName, defaultValue)
{
	var value = getDeviceProperty(getName(), propertyName);
	if ( !(value === "" || value == "undefined") ){
		if ( typeof(defaultValue) == "number" )
			value = Number(value);
		
		setDeviceProperty(getName(), propertyName, value);
		return value;
	}
	
	return defaultValue;
}


function mouseEvent(pressed, x, y, firstPress) {
	if (firstPress)
		setState(state ? 0 : 1);
}

//send captured image file path to registration server
function loop() {
	sendReport();
	delay(1000);
}

//process data received from server
function processData(data, bIsRemote)
{
	if ( data.length <= 0  )
		return;
	data = data.split(",");
	setState(parseInt(data[0]));
}

//send image path to server
var imageLoop=0;
function sendReport()
{
	var report = state + ",";	// comma seperated states
	
	if (state === 0)
		report += '../art/IoE/SmartDevices/camera_off.png';
	else{
		report += '../art/IoE/SmartDevices/camera_image'+imageLoop+'.png';
		imageLoop++;
		if ( imageLoop >= 3)
			imageLoop =0;
	}
	customWrite(0, report);
	IoEClient.reportStates(report);
	setDeviceProperty(getName(), "state", state);
	
}

//set state and update component image to reflect the current state
function setState(newState)
{
	if ( newState === 0 )
		digitalWrite(1, LOW);
	else
		digitalWrite(1, HIGH);
	
	state = newState;
}

File: anmol py/webcam.py
from gpio import *
from time import *
from ioeclient import *
from physical import *
from environment import *
import math

state = 0

def main():
	setup()
	while True:
		loop()
		
#set up client to talk and listen to IoE registration server
def setup():
    global state
    IoEClient.setup({
        "type": "Webcam",
        "states": [{
            "name": "On",
            "type": "bool",
            "controllable": True
        },
        {
            "name": "Image",
            "type": "image"
        }]
    })

    IoEClient.onInputReceive(onInputReceiveDone)
    add_event_detect(0, detect)

    state = restoreProperty("state", 0)
    sendReport()

def onInputReceiveDone(data):
    processData(data, True)
    
def detect():
    processData(customRead(0), False)
    
def restoreProperty(propertyName, defaultValue):
    value = getDeviceProperty(getName(), propertyName)
    if  not (value is "" or value is None):
        if  type(defaultValue) is int :
            value = int(value)

        setDeviceProperty(getName(), propertyName, value)
        return value
    return defaultValue

def mouseEvent(pressed, x, y, firstPress):
    global state
    if firstPress:
        if state == True:
        	setState(0)
        else:
        	setState(1)


#send captured image file path to registration server
def loop():
    sendReport()
    delay(1000)


#process data received from server
def processData(data, bIsRemote):
    if  len(data) <= 0 :
        return
    data = data.split(",")
    setState(int(data[0]))


#send image path to server
imageLoop=0
def sendReport():
    global state
    global imageLoop
    report = str(state) + ","   # comma seperated states

    if state is 0:
        report += '../art/IoE/SmartDevices/camera_off.png'
    else:
        report += '../art/IoE/SmartDevices/camera_image'+ str(imageLoop)+'.png'
        imageLoop = imageLoop + 1
        if  imageLoop >= 3:
            imageLoop =0
    
    customWrite(0, report)
    IoEClient.reportStates(report)
    setDeviceProperty(getName(), "state", state)


#set state and update component image to reflect the current state
def setState(newState):
    global state
    if  newState is 0 :
        digitalWrite(1, LOW)
    else:
        digitalWrite(1, HIGH)

    state = newState
    
if __name__ == "__main__":
    main()